src/DataStructures/Variables.hpp

```diff
@@ -350,13 +351,13 @@ class Variables<tmpl::list<Tags...>> {
   }
   template <typename VT, bool VF>
   friend SPECTRE_ALWAYS_INLINE decltype(auto) operator-(
-      const blaze::Vector<VT, VF>& lhs, const Variables& rhs) noexcept {
-    return ~lhs - rhs.variable_data_;
+      const blaze::DenseVector<VT, VF>& lhs, const Variables& rhs) noexcept {
```

Comment:
Why do you specialize these arguments from `Vector` to `DenseVector`? Shouldn't these operations work fine for any Blaze vector type?

I'm asking because Blaze features such as [`subvector`](https://bitbucket.org/blaze-lib/blaze/wiki/Subvectors) return views into vectors represented as some other vector type in the inheritance hierarchy (I think `DynamicVector`). Specialising these arguments too much would break compatibility with that.

Reply:
My understanding is that these overloads are present primarily to allow operations between `Variables` to work with the expression templates. It does have the upshot of allowing operations between raw blaze types and `Variables`, but I don't think we'd want to encourage that use. If we did want to support vector x Variables operations, we'd want to do so through our own vector types.

Practically, I made this change because it was the most direct way to restore functionality for the expression templates when I changed the underlying vector type to `CustomVector` instead of `PointerVector` -- the expression templates we get from the `CustomVector` would require (at least) a bunch of type-traits like those in `VectorImpl`. That might be the extent of additional changes, but I haven't delved deeply in that direction to work out whether more annoying blaze compilation errors crop up if those are fixed.

Reply:
I believe the change @moxcodes made is correct